interval/interval.go

```diff
@@ -43,8 +55,15 @@ func (i *Interval) RandomInstant() time.Time {
 	end := i.End.Unix()
 	delta := end - start
 
+	// If delta == 0 there is less than 1 second between i.Start and i.End.
 	if delta == 0 {
-		return i.Start
+		// If there is a second-precision time between i.Start and i.End,
+		// return that.
+		if t := time.Unix(end, 0); !t.Before(i.Start) {
```

Comment:
Maybe it's Friday and I'm being slow, but... can you give an example for this. (perhaps you have a test case).

Reply:
so, given that `t` is `end` truncated to second-precision;  and `t` is either equal to `i.Start` or after `i.Start` (which is not truncated so can have a nanos component), then you deem `t` within the interval and return it.

Reply:
And this is the case where `i.End-i.Start` is both sub-second and within the same second.

Reply:
If I got that right, the final statement above seems like a decent comment for it.

Reply:
The logic was wrong again :(  Third time's a charm?